api/autopilot.go

```diff
@@ -54,19 +51,33 @@ type (
 
 	// HostsConfig contains all hosts settings used in the autopilot.
 	HostsConfig struct {
-		AllowRedundantIPs          bool                        `json:"allowRedundantIPs"`
-		MaxDowntimeHours           uint64                      `json:"maxDowntimeHours"`
-		MinProtocolVersion         string                      `json:"minProtocolVersion"`
-		MaxConsecutiveScanFailures uint64                      `json:"maxConsecutiveScanFailures"`
-		ScoreOverrides             map[types.PublicKey]float64 `json:"scoreOverrides"`
+		AllowRedundantIPs          bool   `json:"allowRedundantIPs"`
+		MaxConsecutiveScanFailures uint64 `json:"maxConsecutiveScanFailures"`
+		MaxDowntimeHours           uint64 `json:"maxDowntimeHours"`
+		MinProtocolVersion         string `json:"minProtocolVersion"`
 	}
 )
 
-// EndHeight of a contract formed using the AutopilotConfig given the current
-// period.
-func (ap *Autopilot) EndHeight() uint64 {
-	return ap.CurrentPeriod + ap.Config.Contracts.Period + ap.Config.Contracts.RenewWindow
-}
+var (
+	DefaultAutopilotConfig = AutopilotConfig{
+		Enabled: false,
+		Contracts: ContractsConfig{
+			Amount:      50,
```

Comment:
Probably a conversation worth having while we're refactoring this. Do we need a *max* for number of contracts? The min is easily calculated as something like `minShards*1.5` from the redundancy settings. Having a max makes host selection less efficient, creates churn, and adds complexity since any selection algo must take into account the amount of storage the user needs to spread across a fixed number of target hosts and exclude working hosts that are under that threshold.

Reply:
I think we have tuned this setting a lot in the past. Mostly because more contracts meant potentially more throughput on uploads. 2x should be plenty too though.

Reply:
I'm in favour ✋

Reply:
I think it's a good idea to get rid of it and use 2 times the total shards as per Nate's suggestion. We were planning on changing this to `MinContracts` anyway due to upcoming refactors, which felt a bit awkward anyway. Very easy to reintroduce as well if need be.